### DISCUSSION TITLE
Dynamic themes

### DISCUSSION THREAD

**Discussion by jackdroach** [anuraghazra/github-readme-stats#1415] [Q&A] [answered]
Is there a way to make a card appear with the GitHub light or dark theme depending on the user's settings?

**Comment by lyhsieh** [comment]
No. You should choose your theme manually.

**Reply by anuraghazra** [reply]
GitHub now supports contextual images based on theme. BUT GOD DAMN IT DOESN"T WORK WITH API URLS 😭 

https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#specifying-the-theme-an-image-is-shown-to

![GitHub-Mark-Light](https://user-images.githubusercontent.com/3369400/139447912-e0f43f33-6d9f-45f8-be46-2df5bbc91289.png#gh-dark-mode-only)![GitHub-Mark-Dark](https://user-images.githubusercontent.com/3369400/139448065-39a229ba-4b06-434b-bc67-616e2ed80c8f.png#gh-light-mode-only)

**Reply by lyhsieh** [reply]
so sad QQ

**Comment by idna001** [comment]
Dynamic themes would be very nice..
Isn't there a chance to detect the color of the site?
Or make a theme with colors that look on both themes good and readable, with a clear background.

**Comment by ghost** [comment]
Actually, there is a clever way which was referred in #1268. It makes the background transparent so it looks good on both GitHub themes.

**Reply by Mario-paul** [reply]
Cool solution, thanks! Putting both solutions together makes the readme look really clean.

**Comment by itaditya** [comment, marked as answer]
I've solved it by using the `#gh-dark-mode-only` in anchor href and then place image inside it. Check my [profile readme](https://github.com/itaditya/itaditya/blob/master/README.md)

See it working here-

https://twitter.com/dev__adi/status/1491851695714111488

**Reply by rickstaa** [reply]
As of today, this also is possible using the HTML `<picture>` element in combination with the `prefers-color-scheme` media feature. It is also (for more information, see https://github.blog/changelog/2022-05-19-specify-theme-context-for-images-in-markdown-beta/). Please, however, be aware that in contrast to the method above, this method does not work when people switch the theme without reloading the page.

**Reply by rickstaa** [reply]
Here is a example:


```html
<a href="https://github.com/anuraghazra/github-readme-stats">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=dark">
  <img alt="Shows Anuraghazra's GitHub Stats." src="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=default">
</picture>
</a>
```

<a href="https://github.com/anuraghazra/github-readme-stats">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=dark">
  <img alt="Shows Anuraghazra's GitHub Stats." src="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=default">
</picture>
</a>

**Reply by rickstaa** [reply]
> Here is a example:
> 
> ```
> <a href="https://github.com/anuraghazra/github-readme-stats">
> <picture>
>   <source media="(prefers-color-scheme: dark)" srcset="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=dark">
>   <img alt="Shows Anuraghazra's GitHub Stats." src="https://github-readme-stats.vercel.app/api?username=anuraghazra&theme=default">
> </picture>
> </a>
> ```
> 
> [ ![Shows Anuraghazra's GitHub Stats.](https://camo.githubusercontent.com/f819a99e69cee6e8eb8354df9fa51dbd19592fd8ed699b67cf127d3431194f2e/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170693f757365726e616d653d616e7572616768617a7261267468656d653d64656661756c74) ](https://github.com/anuraghazra/github-readme-stats)

There is one small difference between the GitHub context and GitHub `prefers-colour-scheme` syntax. The first is directly applied when the GitHub theme is changed in realtime while the latter requires a page reload.

**Comment by rickstaa** [comment]
All the currently available ways for creating a dynamic theme are now documented in the [README.md](https://github.com/anuraghazra/github-readme-stats#responsive-card-theme) (See https://github.com/anuraghazra/github-readme-stats#responsive-card-theme).
